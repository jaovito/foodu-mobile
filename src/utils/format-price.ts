import {LanguagePaths} from '@translations/enumns/LanguagePaths';
import {translate} from '@translations/index';

const defaultValues = {
  language: 'en-US',
  currency: 'USD',
};

function getDefaultValues() {
  if (translate(LanguagePaths.GENERAL_LANGUAGE) === 'general.language') {
    return defaultValues;
  }

  return {
    language: translate(LanguagePaths.GENERAL_LANGUAGE),
    currency: translate(LanguagePaths.GENERAL_CURRENCY),
  };
}

export function formatPrice(price: number) {
  return price.toLocaleString(getDefaultValues().language, {
    style: 'currency',
    currency: getDefaultValues().currency,
  });
}
