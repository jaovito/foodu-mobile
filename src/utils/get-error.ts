import {AxiosError} from 'axios';

export function getError(err: Error | unknown): AxiosError {
  if (err instanceof AxiosError) {
    return err;
  }

  return err as AxiosError;
}
