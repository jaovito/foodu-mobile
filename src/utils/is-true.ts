/**
 *
 * Convert booleans values, considering the cases that I understand how true and false.
 * Exemple: Empty arrays and objects, we consider false.
 *
 * @param  param - Can be any value
 *
 * @returns if `value` it isn't an empty array, returns true
 * @returns if `value` it isn't an empty object, returns true
 * @returns if `value` it isn't an empty string, returns true
 * @returns if `value` it isn't null, returns true
 * @returns if `value` it's a non-zero number, returns true
 */
export const isTrue = (param: any): boolean => {
  if (typeof param === 'object') {
    if (param === null) {
      return false;
    } else if (param instanceof Array) {
      return param.length === 0 ? false : true;
    } else {
      return Object.entries(param).length === 0 ? false : true;
    }
  }

  return !!param;
};
