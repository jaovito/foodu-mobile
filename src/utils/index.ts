import {isTrue} from './is-true';
import {formatPrice} from './format-price';

export {isTrue, formatPrice};
