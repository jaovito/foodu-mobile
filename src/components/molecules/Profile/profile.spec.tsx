import {mock} from '@test/mocks';
import {render} from '@test/utils/test-utils';
import {Profile} from './';

describe('<Profile />', () => {
  it('should be defined', () => {
    const profile = render(
      <Profile
        image={mock.profile.image}
        name={mock.profile.name}
        email={mock.profile.email}
      />,
    );

    expect(profile).toBeDefined();
  });

  it('should be able to find the name of the profile', async () => {
    const {findByText} = render(
      <Profile
        image={mock.profile.image}
        name={mock.profile.name}
        email={mock.profile.email}
      />,
    );

    const name = await findByText(mock.profile.name);

    expect(name).toBeDefined();
    expect(name).toBeTruthy();
  });

  it('should be able to find the email of the profile', async () => {
    const {findByText} = render(
      <Profile
        image={mock.profile.image}
        name={mock.profile.name}
        email={mock.profile.email}
      />,
    );

    const email = await findByText(mock.profile.email);

    expect(email).toBeDefined();
    expect(email).toBeTruthy();
  });

  it('should not be able to find the name of the profile', () => {
    const {queryByText} = render(
      <Profile image={mock.profile.image} email={mock.profile.email} />,
    );

    const name = queryByText(mock.profile.name);

    expect(name).toBeFalsy();
  });

  it('should not be able to find the email of the profile', () => {
    const {queryByText} = render(
      <Profile image={mock.profile.image} name={mock.profile.name} />,
    );

    const email = queryByText(mock.profile.email);

    expect(email).toBeFalsy();
  });

  it('should be able to match with snapshot', () => {
    const profile = render(
      <Profile
        image={mock.profile.image}
        name={mock.profile.name}
        email={mock.profile.email}
      />,
    ).toJSON();

    expect(profile).toMatchSnapshot();
  });
});
