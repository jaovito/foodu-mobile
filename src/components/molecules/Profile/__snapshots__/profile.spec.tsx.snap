// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> should be able to match with snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 31,
        "width": "100%",
      },
    ]
  }
>
  <Image
    onError={[Function]}
    resizeMode="cover"
    source={
      {
        "uri": "http://",
      }
    }
    style={
      [
        {
          "borderColor": "#E83F5B",
          "borderRadius": 125,
          "borderWidth": 6,
          "height": 157,
          "width": 157,
        },
      ]
    }
  />
  <View
    style={
      [
        {},
      ]
    }
  >
    <Text
      fontWeight="semiBold"
      numberOfLines={1}
      style={
        [
          {
            "color": "#0D0D0D",
            "fontFamily": "Poppins_600SemiBold",
            "fontSize": 31,
          },
          {
            "fontSize": 31,
          },
        ]
      }
    >
      John Doe
    </Text>
    <Text
      fontWeight="regular"
      numberOfLines={1}
      style={
        [
          {
            "color": "#0D0D0D",
            "fontFamily": "Poppins_400Regular",
            "fontSize": 31,
          },
          {},
        ]
      }
    >
      johndoe@example.com
    </Text>
  </View>
</View>
`;
